tree/dataframe/inc/ROOT/RNTupleDS.hxx

```diff
@@ -53,16 +56,25 @@ class RNTupleDS final : public ROOT::RDF::RDataSource {
       ULong64_t fLastEntry = 0;
    };
 
-   /// The first source is used to extract the schema and build the prototype fields. The page source
-   /// is used to extract a clone of the descriptor to fPrincipalDescriptor. Afterwards it is moved
-   /// into the first REntryRangeDS.
-   std::unique_ptr<Internal::RPageSource> fPrincipalSource;
    /// A clone of the first pages source's descriptor.
    std::unique_ptr<RNTupleDescriptor> fPrincipalDescriptor;
 
    /// The data source may be constructed with an ntuple name and a list of files
    std::string fNTupleName;
    std::vector<std::string> fFileNames;
+   /// Files are opened in the background in batches of size `fNSlots` and kept in the staging area.
+   /// Page sources in the staging area already executed `LoadStructure()`, i.e. they should have a compressed
+   /// buffer of the meta-data available. The page sources in the vector correspond to `fFileNames`.
+   /// Concretely:
+   ///   1. We open the first file on construction to extract the page source and then moved it in the staging area.
```

Comment:
This means that initially the staging area is empty, and after construction only contains the principal page source, is that correct? That seems a bit contradictory to the sentence above ("The page sources in the vector correspond to `fFileNames`"), assuming `fFileNames` has all file names after construction (in case of multiple files).

Reply:
I tried to clarify the comment